### PR TITLE
[findlib] pass findlib flags to the -pack step

### PR DESCRIPTION
**Pull request by gares**
Overlay pr:
- https://github.com/MetaCoq/metacoq/pull/640

### CODE REVIEW

**Comment by gares** [comment]
done

**Review by silene — APPROVED** [review]

**Comment by gares** [comment]
Could you please assign?
Note that not only this CI should be green, but also the one of the overlay PR (which tests the build of single opam packages, not of the entire projects...)

**Review by SkySkimmer — APPROVED** [review]

**Comment by SkySkimmer** [comment]
metacoq not happy

**Comment by silene** [comment]
I suppose that doing `-package coq-metacoq-template.template_coq` is pointless until the package has actually been installed. Perhaps you should keep `-I ../template-coq` around to avoid the issue?

**Comment by gares** [comment]
I think so, since it is the only change of the overlay. I've reverted it, let's see.

**Comment by gares** [comment]
if you install all separate opam files one by one, then `-package` works.
if you compile all with the main makefile, then `-I ../template-coq` is needed since the META is not installed (yet)

**Comment by silene** [comment]
I stroke my suggestion because I feel that you would still need to modify `OCAMLPATH`. Putting back `-I` only helps when going through `coqc`, but not when calling `ocamlopt` directly.

**Comment by gares** [comment]
Sorry, I was messy in my message. The real job that needs to be tested is the bench one (which relies on opam files).
I did test installing the safechecker opam package locally (with this patch and that overlay).

The CI of metacoq itself, I don't know what is wrong.

**Comment by silene** [comment]
> The CI here was broken because of the `-I` which I did remove, and now I did put back.

As I explained, you cannot expect `ocamlfind ocamlc ... whatever.ml` to find `coq-metacoq-template.template_coq` unless it is actually installed or `OCAMLPATH` is modified to point to the non-installed version. Am I missing something?

**Comment by gares** [comment]
That I'm really tired ;-) Thanks for the hint.

**Comment by ejgallego** [comment]
Something you can do and maybe works well for you is actually compute the args to ocamlc directly with ocamlfind (on the right path setup), the put them in the make flags.

**Review by SkySkimmer — APPROVED** [review]

**Comment by mattam82** [comment]
@gares thanks for fixing it, I was wondering what happened to the `opam` CI for metacoq failing on master!

File: dev/ci/ci-metacoq.sh [hunk 1]
```diff
@@ -12,7 +12,6 @@ if [ "$DOWNLOAD_ONLY" ]; then exit 0; fi
 export COQEXTRAFLAGS='-native-compiler no'
 ( cd "${CI_BUILD_DIR}/metacoq"
   [ -e pcuic/metacoq-config ] || ./configure.sh local
-  make .merlin
```
[the remark this answers is not in the document]
Comment (gares) [reply dev/ci/ci-metacoq.sh:15]: It is explained in the commit message.

**Comment by gares** [comment]
Please, double check my comment in the last commit, and maybe guard the `.merlin` rule in safechecker

**Comment by mattam82** [comment]
Yep, .merlin generation is probably not working without a first compile pass, I'll try to fix that.
With your patch we will now get a `requires = "coq-core.plugins.ltac, coq-template-coq.template_ocaml"` in the META file for the safechecker I guess?

**Comment by gares** [comment]
Hum, I have forgot to fix this I'm afraid. I'm a bit puzzled, isn't there an example which uses safecheker?

**Comment by gares** [comment]
let me fix it

**Comment by gares** [comment]
OK re-running ci-metacoq.

Automatic generation of META files only works if there are not external deps.
For it to work one would need a new option to coq_makefile, eg `-findlib-package foo`, and that would also make the line in the `.local` file extending `CAMLPKGS` necessary.

It would be great if someone could implement it.

**Comment by mattam82** [comment]
I can live without the automatic META generation! I suppose safechecker is only loaded after the template plugin in the test-suite/example, so the problem doesn't show up

**Comment by mattam82** [comment]
All good now. If you could please merge this quickly, I'd like to have the CI on metacoq-master back to work.

**Pull request merged**